### PR TITLE
Three players choose word

### LINKED ISSUE #28: As an observer, I can choose the word and be excluded from the game
#5 
Acceptance Tests: Users can choose to play with either 2 players or 3 players. If they choose 3 players, they will then be asked to choose a word. for the other 2 players to compete with.
 If they select 2 players, they will be part of the game in which there is a random word as the word and play against an opponent.

### PR DESCRIPTION
**Pull request by benpalay**
Closes #28

### CODE REVIEW

**Review by IlanMalk — APPROVED** [review]
Awesome work on this one Ben.

There are some additions we should make to this code: 
- When a player creates a 2-player game, it should say `Waiting for other players to join` instead of just showing the room code. 
- The `enter a word` input field is lacking input validation

From an internal quality perspective:
- We should try and find a way to avoid duplicating code between 2 player and 3 player (particularly the keys array and the functions that are common to both). 
- A port isn't needed for testing
- A few extra comments would add clarity (I've highlighted this with my in-line pull request comments)


The pull request is approved though so we can use the code as a basis for the rest of the sprint. ~I've added issues #79 and #80 to remind us to add these features~

File: src/__tests__/socketTest.spec.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    // const IO = io.connect()
+    io.sockets.on('gameCreated', onNewGameCreated)
+
+    function onNewGameCreated (data) {
+      expect(data.GameId).toBe(1000)
+    }
+  })
+})
+const port = process.env.PORT || 3000
```
[the remark this answers is not in the document]
Comment (benpalay) [reply src/__tests__/socketTest.spec.js:31]: Fixed. Thank you

File: src/public/playMulti.js [hunk 2]
```diff
@@ -253,60 +295,119 @@ const App = {
...
+                  if (wordOfTheDay === 'word of the day') {
+                    const data = {
+                      myRole: App.myRole,
+                      gameId: App.gameId
+                    }
+                    feedbackForGuess('Correct')
+                    isGameEnded = true
+                    IO.socket.emit('gameWinner', data)
+                  } else {
+                    if (currentRow === 5) {
```
[the remark this answers is not in the document]
Comment (benpalay) [reply src/public/playMulti.js:382]: Thanks. Good catch. However it needs to be checking if its the right word after 5 inputs, so the double else-if would potentially negate that.

File: src/public/playMulti.js [hunk 3]
```diff
@@ -410,8 +504,36 @@ const App = {
       ]
       let currentRow = 0
       let currentTile = 0
-
-      // public
+      const keys = [
```
[the remark this answers is not in the document]
Comment (benpalay) [reply src/public/playMulti.js:507]: Agreed. Potential issues with each player accessing a technically separate html doc and so it may not work.

**Pull request merged**